packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,135 @@
+import { glob } from 'glob';
+import { BackendIdentifier } from '@aws-amplify/plugin-types';
+import {
+  CloudFormationClient,
+  DeleteStackCommand,
+} from '@aws-sdk/client-cloudformation';
+import { BackendIdentifierConversions } from '@aws-amplify/platform-core';
+import { getClientConfigPath } from '@aws-amplify/client-config';
+import { TestBranch, amplifyAppPool } from '../amplify_app_pool.js';
+import { e2eToolingClientConfig } from '../e2e_tooling_client_config.js';
+import { setupPackageManager } from './setup_package_manager.js';
+
+void describe('installs expected packages and scaffolds expected files', async () => {
+  let branchBackendIdentifier: BackendIdentifier;
+  let testBranch: TestBranch;
+  let cfnClient: CloudFormationClient;
+  let tempDir: string;
+  let packageManagerExecutable: string;
+  let packageManager: string;
+
+  beforeEach(async () => {
+    tempDir = await fsp.mkdtemp(path.join(os.tmpdir(), 'test-create-amplify'));
+    const packageManagerInfo = await setupPackageManager(tempDir);
+    packageManagerExecutable = packageManagerInfo.packageManagerExecutable;
+    packageManager = packageManagerInfo.packageManager;
+
+    // start a local npm proxy and publish the current codebase to the proxy
+    await execa('npm', ['run', 'clean:npm-proxy'], { stdio: 'inherit' });
+    await execa('npm', ['run', 'vend'], { stdio: 'inherit' });
+
+    cfnClient = new CloudFormationClient(e2eToolingClientConfig);
+    testBranch = await amplifyAppPool.createTestBranch();
+    branchBackendIdentifier = {
+      namespace: testBranch.appId,
+      name: testBranch.branchName,
+      type: 'branch',
+    };
+  });
+
+  afterEach(async () => {
+    await cfnClient.send(
+      new DeleteStackCommand({
+        StackName: BackendIdentifierConversions.toStackName(
+          branchBackendIdentifier
+        ),
+      })
+    );
+    await fsp.rm(tempDir, { recursive: true });
+    // stop the npm proxy
+    await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
+  });
+
+  void it(`starting project`, async () => {
+    await execa(
+      packageManager.startsWith('yarn') ? 'yarn' : packageManager,
+      ['create', 'amplify', '--yes'],
+      {
+        cwd: tempDir,
+        stdio: 'inherit',
+      }
+    );
+
+    const amplifyPathPrefix = path.join(tempDir, 'amplify');
+
+    const pathPrefix = path.join(tempDir, 'amplify');
+
+    const files = await glob(path.join(amplifyPathPrefix, '**', '*'), {
+      nodir: true,
+      windowsPathsNoEscape: true,
+      ignore: ['**/node_modules/**', '**/yarn.lock'],
+    });
+
+    const expectedAmplifyFiles = [
+      path.join('auth', 'resource.ts'),
+      'backend.ts',
+      path.join('data', 'resource.ts'),
+      'package.json',
+      'tsconfig.json',
+    ];
+
+    assert.deepStrictEqual(
+      files.sort(),
+      expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
+    );
+
+    // assert that project compiles successfully
+    await execa(
```

Comment:
pipeline-deploy will also run tsc. Could we just do pipeline-deploy and take this out?

Reply:
Removed 17f406b70c69579ad2aa87c69f57018c8c2dd2f9